regimes/pt/pt.go

```diff
@@ -41,6 +49,23 @@ const (
 	ZoneMadeira        l10n.Code = "30" // Autonomous Region
 )
 
+// Additional zone definition for mainland Portugal
+const (
+	ZoneMainland l10n.Code = "00"
```

Comment:
Here I'm adding a new super-zone that includes all mainland PT (i.e. everything except the autonomous region). I put it in a different const section than the other zones because I'm not using an ISO-3166-2:PT but a custom code. The idea is to make it the default zone. And that way get access to the zone metadata and get its `TaxCountryCode` when needed.

Would it make sense to you if we set this zone automatically in the `calculate()` function when no other zone is given?

Reply:
I think `calculate()` is a good place to inject default values when absent. Do we want to distinguish between the situation where the user intentionally left it out and a case where user forgot to select the zone (in which case I guess the validation should fail)?

Reply:
Right! In this case, I think it's clear to me that the zone is not really optional. All tax reports in Portugal require to specify the `TaxCountryRegion` which possible values are (PT, PT-AZ, PT-MA). That's why I think it would make sense to make this field mandatory in PT and, for convenience, inject a "00" (i.e. Portugal Mainland) value when nothing else is provided. But perhaps we prefer users to be explicit about this?

Reply:
That makes sense to me!

Reply:
Yes, I made up the "00" code as there's no ISO-3166 code explicitly for mainland Portugal. You either have 'PT' which represents the whole country or 'PT-01'...'PT-18' for each district in mainland Portugal.

Now, I see your concerns. As I explained in my [general comment](https://github.com/invopop/gobl/pull/114#issuecomment-1483117130), my goal here was being able to map directly from a GOBL's tax zone to an AT's tax country region using metadata in the regime file (and avoid users to specify district zones in PT mainland which is fiscally irrelevant). But we don't really any of this, we can keep things simple here and do [all the mapping in the provider](https://github.com/invopop/invoicexpress/blob/main/app/domain/invoicexpress/payload_builder.rb#L142-L145). So, I'll dismiss this part of the PR.